src/Repository/HealthCheckConfigRepository.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace PrestaShop\Module\HealthCheck\Repository;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Statement;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\Module\HealthCheck\Entity\HealthCheckConfig;
+use PrestaShop\PrestaShop\Core\Exception\DatabaseException;
+
+class HealthCheckConfigRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $tableName;
+
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix
+    ) {
+        $this->connection = $connection;
+        $this->tableName = sprintf('%shealth_check_config', $dbPrefix);
+    }
+
+    public function getLastHealthCheckConfig(): ?HealthCheckConfig
+    {
+        /** @var QueryBuilder $qb */
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('id_health_check_config as id, token, ips')
+            ->from($this->tableName)
+            ->where(sprintf('id_health_check_config = (SELECT MAX(id_health_check_config) FROM %s)', $this->tableName))
+        ;
+
+        $data = $qb->execute()->fetchAll(FetchMode::ASSOCIATIVE);
+
+        if (!is_array($data) || empty($data) || empty($data[0])) {
+            return null;
+        }
+
+        return (new HealthCheckConfig())->hydrate($data[0]);
+    }
+
+    /**
+     * @throws DatabaseException
+     */
+    public function update(array $data)
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->update($this->tableName)
+            ->set('token', ':token')
+            ->set('ips', ':ips')
+            ->setParameters([
+                'token' => $data['token'],
+                'ips' => $data['ips'],
+            ])
+        ;
+        $this->executeQueryBuilder($qb, 'HealthCheck config error');
+    }
+
+    /**
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function createTables(): array
+    {
+        $this->dropTables();
+
+        $query = sprintf(
+            'CREATE TABLE IF NOT EXISTS `%s`(
+`id_health_check_config` int(10) unsigned NOT NULL auto_increment,
```

Comment:
Is it really needed, because of Doctrine we should let it create the database :thinking:

Reply:
I tested. Doctrine doesnt do it by its own